pydatastructs/linear_data_structures/linked_lists.py

```diff
@@ -581,3 +583,133 @@ def extract(self, index):
         elif index == 0:
             self.tail.next = self.head
         return node
+
+class SkipList(object):
+    """
+    Represents Skip List
+
+    Examples
+    ========
+
+    >>> from pydatastructs import SkipList
+    >>> sl = SkipList()
+    >>> sl.insert(6)
+    >>> sl.insert(1)
+    >>> sl.insert(3)
+    >>> repr(sl)
+    '-inf.->1.->3.->6.->inf.'
+    >>> sl.remove(1)
+    >>> sl.insert(4)
+    >>> sl.insert(2)
+    >>> sl.search(4)
+    True
+    >>> sl.search(10)
+    False
+    >>> repr(sl)
+    '-inf.->2.->3.->4.->6.->inf.'
+
+    """
+
+    def __new__(cls):
+        obj = object.__new__(cls)
+        obj.head = None
+        obj.tail = None
+        obj._addOneLevel()
+        return obj
+
+    def __repr__(self):
+        li = self.__str__()
+        return '->'.join(li[-1])
+
+    def __vars__(self):
+        li = self.__str__()
+        return ' '.join(map(lambda l: '->'.join(l), li))
+
+    def __str__(self):
+        li = []
+        node = self.head
+        while node:
+            l = []
+            no = node
+            while no:
+                l.append('{}{}'.format(no.val, '' if no.down else '.'))
+                no = no.next
+            li.append(l)
+            node = node.down
+        return li
+
+    def _addOneLevel(self):
+        self.tail = SkipNode(math.inf, None, self.tail)
+        self.head = SkipNode(-math.inf, self.tail, self.head)
+
+
+    def _growUp(self):
```

Comment:
Please write as `_grow_up` ([snake case](https://en.wikipedia.org/wiki/Snake_case)).

Reply:
yep will make the changes